### ISSUE TITLE
False positive for `eslint-plugin-unicorn(require-post-message-target-origin)`

### ISSUE THREAD

**Issue by jpzwarte** [oxc-project/oxc#12026]
```
  ⚠ eslint-plugin-unicorn(require-post-message-target-origin): Missing the `targetOrigin` argument.
     ╭─[packages/web-shell/src/app/app.ts:116:77]
 115 │       if (event.data.action === 'getAccessToken') {
 116 │         event.ports[0].postMessage({ accessToken: this.#user?.access_token });
     ·                                                                             ─
 117 │       }
     ╰────
  help: Insert `, self.location.origin`
```

The above is incorrect. `postMessage` is not called on `window` here, but `MessagePort`. See https://developer.mozilla.org/en-US/docs/Web/API/MessagePort/postMessage

Similar, but not the same as #10106

**Comment by jpzwarte** [comment]
Also see https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1396#issuecomment-878077241

**Comment by camc314** [comment]
Related #11638

**Comment by ImanKCO** [comment]
Is this released? I have the same issue on `1.23.0` currently latest. 
I'm doing broadcast channel post messages and that does not have 2 arguments but oxlit wants me to set the origin on that which is wrong.

**Comment by reana-ovo** [comment]
I have experienced this issue on `1.34.0`, seems that this problem still hasn't been solved :(

**Comment by camc314** [comment]
> I have experienced this issue on `1.34.0`, seems that this problem still hasn't been solved :(

Can you share a repro? I tried this in the playground and I couldn't reproduce it:

https://playground.oxc.rs/#eNptVMFu2zAM/ZVApw1Itq7AgMI7De0CDGjXbi3Ww7KDLNOuWll0SbpJYeTfRzm22gFFDpYeRfLxkcxgnCkMPEGUDx2S8J+jv3pguQBm28C7YWGd0/MNPkAsFhuzMYv9+y+bmH5bHyvc/ve+PXzfm6VBUwyG+pg+wUcxhVAPS1MjtVYEyBS1DayIkI2c4Ix4xmAFqjNwwZIVj5GzcXvnBbizDjLU2tiEl6vDtiNlkgF22Kn5QICf2xLDfHN1Mz3bL01niROxwcBOILLmVX2Ed1qPDQG3v0B6ipe9sK9g3UeXqM2hUk6gJ7iypL4zOvr9PvkehbxGdNnAoLTFu29ESBOqHJJWBw4OY+2VnRlUeOoD8MYUi3Tpo3dI8SPBY+8JVqkFq0n7lVhqQFZIvvExeWwMpBSpdYu90RSvWjCYnuHGli9aiS1vfSV3pjheGojVZX3uo4pnQq0qdFqETPaTIy3CJ+V/9igv8t/z7voN+DHdrgg7zWUsryJABZXG1P57rbo5xba1yRaCoiqPzxoq/+0sq9q39pn1SUnWPYBc6yyo+/x4Rm0LB+ZTfizvwckt2U5jzL1KiUauX0X7U/bKEOiVm6qVx/Og10HetDZ8fPTpswZQBc+gVp810mmwzGsPocp9hl0H5FvdMRt0oFHnOfdbra2XjF6A2MqKijANw9uLMBhWsp2OFFC0OsozVYcVNDCuXEw9zlN+zxXqqk9qxogyxpqRAM38dowyr8+gt3m3tHLszvW/Igd9AOh+qMrz7IyuOuYY1jryyeMJqER+paWPqQfJlE97hatRvgTn0z7hSea5pv0/3varQw==

**Comment by reana-ovo** [comment]
> > I have experienced this issue on `1.34.0`, seems that this problem still hasn't been solved :(
> 
> Can you share a repro? I tried this in the playground and I couldn't reproduce it:
> 
> https://playground.oxc.rs/#eNptVMFu2zAM/ZVApw1Itq7AgMI7De0CDGjXbi3Ww7KDLNOuWll0SbpJYeTfRzm22gFFDpYeRfLxkcxgnCkMPEGUDx2S8J+jv3pguQBm28C7YWGd0/MNPkAsFhuzMYv9+y+bmH5bHyvc/ve+PXzfm6VBUwyG+pg+wUcxhVAPS1MjtVYEyBS1DayIkI2c4Ix4xmAFqjNwwZIVj5GzcXvnBbizDjLU2tiEl6vDtiNlkgF22Kn5QICf2xLDfHN1Mz3bL01niROxwcBOILLmVX2Ed1qPDQG3v0B6ipe9sK9g3UeXqM2hUk6gJ7iypL4zOvr9PvkehbxGdNnAoLTFu29ESBOqHJJWBw4OY+2VnRlUeOoD8MYUi3Tpo3dI8SPBY+8JVqkFq0n7lVhqQFZIvvExeWwMpBSpdYu90RSvWjCYnuHGli9aiS1vfSV3pjheGojVZX3uo4pnQq0qdFqETPaTIy3CJ+V/9igv8t/z7voN+DHdrgg7zWUsryJABZXG1P57rbo5xba1yRaCoiqPzxoq/+0sq9q39pn1SUnWPYBc6yyo+/x4Rm0LB+ZTfizvwckt2U5jzL1KiUauX0X7U/bKEOiVm6qVx/Og10HetDZ8fPTpswZQBc+gVp810mmwzGsPocp9hl0H5FvdMRt0oFHnOfdbra2XjF6A2MqKijANw9uLMBhWsp2OFFC0OsozVYcVNDCuXEw9zlN+zxXqqk9qxogyxpqRAM38dowyr8+gt3m3tHLszvW/Igd9AOh+qMrz7IyuOuYY1jryyeMJqER+paWPqQfJlE97hatRvgTn0z7hSea5pv0/3varQw==

Just call `postMessage` method on a random object than it will complain.

https://playground.oxc.rs/#eNptU8Fu2zAM/ZVApw1Itq7AgCI7De0CDGjXbh3WSy60TLtqZckl6SaFkX8f5VhqD0UOsR5F8vHxaTTWrA186iPLFTJDix+64//Hb9uQfjsX6rh794ZZmmjWo6EhpD/vgpi10IBL00TqQATJrBvwrIgQBE5wQRxHD4L1BVoPBOJi4BLc3TtB7sFigToIrX892tj1pEwKwDb2Gj4S4Jeuij6fbNPO1w5L0wNxIjYa3AsG1r4qgvBe5wHv4+4PykDhehB2NW6GYBO1XCr1RHrGGyDNzeiU9+/sZxByWtGWAKPSFmd/EEWaUeWQtDpysDE0TtmZcbFVJT3y1qwX6TAEZyOFz4RPgyNcpRWsZu1XAtSirCK51oWUsTWYWmzN4rA4GG3xZgWjGRj/QvWqlUB152q5N+vTpcFQXzeXLqh4xjeqQq9DyBw/O9EhXFL+9xDlVf4H3t++Az+l0w3FXnsZ4FVArLHWmrp/p1O357HrIMW8V1TlcUVD5b/Lsmp8By+sVyoC+4hyq17Q9Hw5o9DhkfncP1YPaOWOoNcaeVep0cT1u+h+qkEZIr1JU7WKPY96HeXVEsinJ1++agFV8AIbzdlEOvfAvHHo67Jn3PdIrsMg4NXQUf1c9q3RzklBr1CgBlERZjO8/xBGw0q2V0shBVArZ6o21tji9ORC2nFx+QPXUd/zrGYIUaZaGfHY5rtTlfx8Rj3lt6WTx/4Sn7EUfUTsf6nK2TtTqto8+o1aPmU8I1WR32jpQtpBCpWvg8L1JF+Cy9ch4UnmPNPhP17Mohg=

**Comment by wuzzeb** [comment]
I am still getting the false positive for broadcast channel too, and looking at the supposed fix https://github.com/oxc-project/oxc/pull/14192/changes it doesn't check for any of that.  The pull request 14192 just checks if the stuff before the call includes the word "port".  So really, 14192 may fix one false positive but by no means fixes them all.

So I don't know if this should be reopened or a new issue filed referencing this one?
